### PR TITLE
accept forced physics results from entity server (e.g., when filtered)

### PR DESCRIPTION
**Pull request by howard-stearns**
test plan:
- Try the following two filters: [sandbox-zone-filter](https://s3-us-west-1.amazonaws.com/hifi-content/zfox/sandbox-filter.js) and [maxVelocity-filter](https://s3-us-west-1.amazonaws.com/hifi-content/zfox/maxVelocity-filter.js). Use the test plan from https://github.com/highfidelity/hifi/pull/9558. Those two filters should work at least as well as in #9558.
- Now, try the following two filters, which are slightly modified versions of the above filters: [sandbox-zone-filter-noRandomness](https://s3-us-west-1.amazonaws.com/hifi-content/zfox/sandbox-filter-noRandomness.js) and [maxVelocity-filter-noRandomness](https://s3-us-west-1.amazonaws.com/hifi-content/zfox/maxVelocity-filter-noRandomness.js). Use the test plan from #9558.
Does it still work? (Compare to not using the client from this PR. Without the random, we expect the master client to not be good.)
- Without any filters, makes sure that physics and near/far grab work well, especially when there are multiple people. Especially:
 -- Turn on physics ownership markers, and move a physical object within a pile of them, with multiple people present. Make sure the objects settle as they are supposed to, and don't jump unexpectedly.
 -- Use physical toys such as planky, bow and arrow, ping pong. Make sure it's as good as master.
- Try the following filter (either for a specific region or everywhere).  You should be able to interact with objects physically, but not make any other changes. (Near/far grabbing becomes unsatisfying. It "works" only to the extent that it looks like physics. The near/far gab itself is countermanded.

### CODE REVIEW

**Comment by howard-stearns** [comment]
Note: this pr changes the client, but not the entity server. There is no protocol change. So you can use this client with dev-demo, dev-playa, etc.

**Comment by hifi-gustavo** [comment]
Build console output for commit: fe8fe816f4aa0b83e76d4f967e58fcb16798e83b available at: https://gist.github.com/ba955ec29cef2e431fd4160537c1959b<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9560-fe8fe81.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9560-fe8fe81.dmg

File: libraries/entities/src/EntityItem.cpp [hunk 1]
```diff
@@ -708,54 +710,54 @@ int EntityItem::readEntityDataFromBuffer(const unsigned char* data, int bytesLef
...
                 _lastUpdatedAngularVelocityValue = value;
             }
         };
 
-        auto customSetAcceleration = [this, lastEdited, overwriteLocalData, weOwnSimulation](glm::vec3 value){
+        auto customSetAcceleration = [this, lastEdited, overwriteLocalData, weOwnSimulation, filterRejection](glm::vec3 value){
             bool simulationChanged = lastEdited > _lastUpdatedAccelerationTimestamp;
             bool valueChanged = value != _lastUpdatedAccelerationValue;
-            bool shouldUpdate = overwriteLocalData && !weOwnSimulation && simulationChanged && valueChanged;
+            bool shouldUpdate = overwriteLocalData && !weOwnSimulation && simulationChanged && (valueChanged || filterRejection);
```
Comment (davidkelly) [inline libraries/entities/src/EntityItem.cpp]: I wonder if we can DRY this up -- the simulationChanged, valueChanged and shouldUpdate in the 5 spots above are so close to being the same.  Nothing worth holding it up, just a thought.

**Comment by davidkelly** [comment]
testing now -- but the last point in the test plan mentions a filter that isn't linked.

**Comment by hifi-gustavo** [comment]
Build console output for commit: 8d8b338c66b0634befe5a03c3d8c5e80474fcafb available at: https://gist.github.com/ba955ec29cef2e431fd4160537c1959b<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9560-8d8b338.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9560-8d8b338.dmg

File: libraries/entities/src/EntityItem.cpp [hunk 2]
```diff
@@ -663,6 +664,7 @@ int EntityItem::readEntityDataFromBuffer(const unsigned char* data, int bytesLef
         if (wantTerseEditLogging() && _simulationOwner != newSimOwner) {
             qCDebug(entities) << "sim ownership for" << getDebugName() << "is now" << newSimOwner;
         }
+        filterRejection = newSimOwner.getID().isNull();
```
Comment (ZappoMan) [inline libraries/entities/src/EntityItem.cpp]: this probably deserves a comment... why is a "null" sim owner ID a filter rejection....

Comment (howard-stearns) [inline libraries/entities/src/EntityItem.cpp]: good idea

Comment (howard-stearns) [inline libraries/entities/src/EntityItem.cpp]: done

**Comment by hifi-gustavo** [comment]
Build console output for commit: 18f8c2b866e3aaa4d640cd450f7ac67967d2d7f8 available at: https://gist.github.com/ba955ec29cef2e431fd4160537c1959b<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9560-18f8c2b.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9560-18f8c2b.dmg

**Comment by howard-stearns** [comment]
test this please

**Comment by hifi-gustavo** [comment]
Build console output for commit: 6f4441028d5f5328bfe6e184bb0bc2f4fca23eaa available at: https://gist.github.com/ba955ec29cef2e431fd4160537c1959b<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9560-6f44410.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9560-6f44410.dmg

**Comment by zfox23** [comment]
QA Pass, but re: the maxVelocity filter:
@davidkelly and I were curious about the intent of the maximum velocity filter. I'm wondering if we should set the maximum velocity in a production filter to something higher than 5m/s, such as 50 or 100 m/s. Setting the maxVelocity to 5m/s means that far grabbing entities is going to incur a very poor UX.

**Pull request merged**